src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -96,6 +109,9 @@ def collect_package_tools(keys: Optional[List[str]] = None) -> dict:
                 importlib.import_module(m)  # Import the module to make sure it is valid
                 tool["package"] = entry_point.dist.metadata['Name']
                 tool["package_version"] = entry_point.dist.version
+                # Set default input index to ui_hints
```

Comment:
Why should we set the default index?

Reply:
I think such ui hints would be better set in the yaml instead of here.

Reply:
We set the default index because customers complained that the order of their custom-llm tool inputs displayed in the UI was inconsistent with the order defined in the yaml file. 
We have another[ pr set index in the yaml](https://github.com/microsoft/promptflow/pull/1737), but after internal discussion, we think it is better set default index.